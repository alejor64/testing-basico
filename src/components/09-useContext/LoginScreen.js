import React, { useContext } from 'react'
import { UserContext } from './UserContext'

const LoginScreen = () => {

    const {setUser} = useContext(UserContext)

    const object = {
        id: 123,
        name: 'Alejandro'
    }

    return (
        <div>
            <h1>Login Screen</h1>
            <hr />

            <button
                className="btn btn-primary"
                onClick={() => setUser(object)}
            >
                Login
            </button>
        </div>
    )
}

export default LoginScreen
