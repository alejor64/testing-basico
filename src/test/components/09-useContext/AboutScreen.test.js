import React from 'react'
import '@testing-library/jest-dom'
import {mount} from 'enzyme'
import AboutScreen from '../../../components/09-useContext/AboutScreen'
import { UserContext } from '../../../components/09-useContext/UserContext'

describe('Prueba en el componente <AboutScreen />', () => {

    const user = {
        name: 'Alejandro',
        email: 'alejoruizs@hotmail.com'
    }
    const setUser = jest.fn()

    const wrapper = mount(
        <UserContext.Provider value={{
            user,
            setUser
        }}>
            <AboutScreen />
        </UserContext.Provider>
    )

    test('Debe rendizar el componente', () => {
        expect(wrapper).toMatchSnapshot()
    })
    
})
