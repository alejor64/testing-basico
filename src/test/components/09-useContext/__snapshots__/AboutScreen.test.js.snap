// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en el componente <AboutScreen /> Debe rendizar el componente 1`] = `
<div>
  <h1>
    About Screen
  </h1>
  <hr />
  <pre>
    {
   "name": "Alejandro",
   "email": "alejoruizs@hotmail.com"
}
  </pre>
  <button
    className="btn btn-warning"
    onClick={[Function]}
  >
    LongOut
  </button>
</div>
`;
